src/main/java/org/ibp/api/rest/keysequenceregister/KeySequenceResource.java

```diff
@@ -25,10 +25,11 @@
 	private KeySequenceRegisterService keySequenceRegisterService;
 
 	@ApiOperation(value = "Delete Key Prefix Cache")
-	@PreAuthorize("hasAnyAuthority('ADMIN', 'GERMPLASM', 'MANAGE_GERMPLASM')")
-	@RequestMapping(value = "/crops/{cropName}/key-sequences", method = RequestMethod.DELETE)
+	@PreAuthorize("hasAnyAuthority('ADMIN', 'GERMPLASM', 'MANAGE_GERMPLASM', 'DELETE_GERMPLASM')")
+	@RequestMapping(value = "/crops/{cropName}/programs/{programUUID}/key-sequences", method = RequestMethod.DELETE)
```

Comment:
`programUUID` should not be part of the path variable as key sequences are not program-specific resources. Better specify it as optional request parameter (same as with germplasm resources):
`@RequestParam(required = false) final String programUUID`